### ISSUE TITLE
optimizer selection question

### ISSUE THREAD

**Issue by acse-wz19** [ar4/deepwave#54]
Hi Alan,

For least-squares RTM, when I choose the SGD optimizer as the examples you show in the document, I can get the correct image. However, if I choose to use the Adam optimizer with lr=10 the image looks bad. Here, I post the image of SGD and Adam separately. Do you have any suggestions for this situation if I want to generate the same image quality as SGD when I choose Adam?
![image](https://user-images.githubusercontent.com/56032987/227982980-33dcd7d2-4dcb-481c-a67e-92d4c1ebf2b5.png)
![image](https://user-images.githubusercontent.com/56032987/227983127-f8aa0999-8994-4e40-9052-6809b1f0c10b.png)

**Comment by ar4** [comment]
Hello,

Is the second image the one produced with Adam? Depending on how smooth
your velocity model is, the second one might actually be a more accurate
LSRTM result. You could check how accurate both of them are by forward
propagating through both of the images using scalar_born (with scatter set
as the image) and comparing the output with the true data.

Adam behaves very differently to SGD. SGD uses the gradient amplitude to
update the model/image, which is why I need to scale it in the examples to
get the update amplitude to an appropriate value (which may need to be
adjusted depending on the data). You can try running one epoch, and then
plotting scatter.grad.detach() with the colorbar turned on to look at the
amplitudes of the gradient. When using SGD you will want to adjust the
scaling applied to the loss, or the SGD learning rate, so that when the
gradient is scaled by this value it will be of an appropriate amplitude
(probably a maximum of less than 100). Adam uses the sign of the gradient.
This will cause it to make changes in parts of the model/image even where
the gradient is small and SGD would have made a negligible change there. To
try to get a result closer to SGD, you should thus probably use a smaller
Adam learning rate. As I said above, though, it may be that Adam is doing a
better job and it is the SGD result that needs to improve by running for
more iterations.

**Comment by acse-wz19** [comment]
yes, the second one is the results of adam. and thanks for your answer.

**Comment by ar4** [comment]
I am going to close this Issue, but please feel free to reopen it if you have another question about this.
